### ISSUE TITLE
Defaults aren't respected if empty store is passed.

### ISSUE THREAD

**Issue by DRBragg** [palkan/store_attribute#45]
### Tell us about your environment

**Ruby Version:** 3.3.4

**Rails Version:** 7.1.3.4

**PostgreSQL Version:** 15.4

**Store Attribute Version:** Attempting to update to 1.3 from 0.9.3

### What did you do?
 Perviously we could do something like:
```ruby
class Pages::PageSection < ApplicationRecord
  store_attribute :content, :heading_level, :string, default: "1"
end
```

and get

```ruby
ps = Pages::PageSection.new(content: {})
ps.heading_level
#=> "1"
```
but after attempting to upgrade we're now seeing this:
```ruby
ps = Pages::PageSection.new(content: {})
ps.heading_level
#=> nil
```

### What did you expect to happen?

We expected the original behavior to continue. We also attempted to set `store_attribute_unset_values_fallback_to_default` to `true` in `ApplicationRecord` with no change in behavior

### What actually happened?
Default value are no longer being set if an empty store is passed.

**Comment by palkan** [comment]
Based on the current design, it's an expected behaviour: defaults are only applied to the store as a whole on write (so, providing an empty value wouldn't trigger defaults); and the `store_attribute_unset_values_fallback_to_default` is only used during the deserialize (from the database) phase.

I wonder though if we should make `store_attribute_unset_values_fallback_to_default` work for user provided values, too 🤔 That sounds like a solution to the problem. WDYT?

**Comment by DRBragg** [comment]
> Based on the current design, it's an expected behaviour: defaults are only applied to the store as a whole on write (so, providing an empty value wouldn't trigger defaults); and the `store_attribute_unset_values_fallback_to_default` is only used during the deserialize (from the database) phase.
> 
> I wonder though if we should make `store_attribute_unset_values_fallback_to_default` work for user provided values, too 🤔 That sounds like a solution to the problem. WDYT?

Our issue is the change in behavior so if that would fix it then we're all for it. 😅

**Comment by palkan** [comment]
Ok, should be fixed on master. Could you check, please?

**Comment by DRBragg** [comment]
> Ok, should be fixed on master. Could you check, please?

Looks good, @palkan! Thanks!!

**Comment by ermolaev** [comment]
Hello, I updated to 1.3.1, and have this issue


```ruby
ActiveRecord::Schema.define do
  create_table :models, force: true  do |t|
    t.jsonb :settings
  end
end

class Model < ApplicationRecord
  store_attribute :settings, :name, :string, default: "T"
end

StoreAttribute::VERSION # => "1.3.1
Model.create(settings: {}).name # => nil, should "T"
```

**Comment by ermolaev** [comment]
@palkan, what do you think if `self.store_attribute_unset_values_fallback_to_default = true` set by default in gem

this behavior seems more intuitive, and not everyone devs reads the documentation to the end 😀

also, other gem [attr_json](https://github.com/jrochkind/attr_json) [jsonb_accessor](https://github.com/madeintandem/jsonb_accessor) have it by default

**Comment by palkan** [comment]
I think, that's something worth considering for the next major release. Let's wait for Rails 8 final version to come out, so we can do that.

**Comment by palkan** [comment]
> what do you think if self.store_attribute_unset_values_fallback_to_default = true set by default in gem

Changed and released in 2.0.0
